### PR TITLE
ISSUE-856 Added SetWithPath cookie method

### PR DESCRIPTION
**Pull request by orderbynull**
This PR refers issue https://github.com/gobuffalo/buffalo/issues/856

### CODE REVIEW

File: cookies_test.go [hunk 1]
```diff
@@ -40,6 +40,18 @@ func TestCookies_Set(t *testing.T) {
...
+	res := httptest.NewRecorder()
+
+	c := Cookies{&http.Request{}, res}
+
+	c.SetWithPath("name", "Rob Pike", "/foo")
+
+	h := res.Header().Get("Set-Cookie")
+	r.Equal("name=\"Rob Pike\"; Path=/foo", h)
+}
+
```
Comment (robbyoconnor) [inline cookies_test.go]: Shouldn't you also test with the "/" path to ensure that this cookie is not set for that?

Comment (robbyoconnor) [inline cookies_test.go]: Shouldn't you also test with the `/` path to ensure that this cookie is not set for that?

Comment (orderbynull) [reply cookies_test.go]: Not sure what do you mean. If i set c.SetWithPath for path '/' it will set.

**Comment by markbates** [comment]
Thanks!

**Pull request merged**